app/adapters/vhost.js

```diff
@@ -1,8 +1,31 @@
 import ApplicationAdapter from './application';
 import buildURLWithPrefixMap from '../utils/build-url-with-prefix-map';
+import Ember from 'ember';
 
 export default ApplicationAdapter.extend({
   buildURL: buildURLWithPrefixMap({
     'services': 'service.id'
-  })
+  }),
+
+  // https://github.com/emberjs/data/blob/64ccb96673b2396dc16424b8ab1271afa43134ca/packages/ember-data/lib/adapters/rest_adapter.js#L549
+  // To update a VHost, the adapter should send a PUT request to /vhosts/:id,
+  // so we override updateRecord and remove the prepended /services/:service_id
+  // that `buildURL` adds
+  updateRecord: function(store, type, record){
+    var data = {};
+    var serializer = store.serializerFor(type.typeKey);
+
+    var snapshot = record._createSnapshot();
+    serializer.serializeIntoHash(data, type, snapshot);
+
+    var id = Ember.get(record, 'id');
+
+    var serviceId = Ember.get(record, 'service.id');
+    var url = this.buildURL(type.typeKey, id, record);
+    if (serviceId) {
+      url = url.replace("/services/" + serviceId, "");
+    }
+
+    return this.ajax(url, "PUT", { data: data });
+  }
```

Comment:
You may be right — I'm not sure how best to generalize this though, because not all of the models that we send updates on have anything prepended to them.

Looking through the test suite, the only PUTs we do at the moment are `/users/:user_id`, `/services/:service_id` and `/vhosts/:vhost_id`. The first two of those do not have anything prepended to their urls, but the latter does.

The only time we call `DELETE` in the test suite at the moment is for the ssh keys, so there would be no harm in generalizing the `destroyRecord` override (but also no gain)